splunk_connect_for_snmp_poller/manager/static/mib_enricher.py

```diff
@@ -51,15 +50,25 @@ def extract_dimension_name_and_value(dimension, index):
 class MibEnricher:
     def __init__(self, mib_static_data_collection):
         self._mib_static_data_collection = mib_static_data_collection
+        self.dimensions_fields = self.__collect_if_mib_fields(
+            mib_static_data_collection
+        )
+
+    def __collect_if_mib_fields(self, mib_static_data_collection):
+        fields = []
+        if not mib_static_data_collection:
+            return []
+        for el in mib_static_data_collection:
+            fields += list(el.keys())
+        logger.info(f"_mib_static_data_collection: {mib_static_data_collection}")
+        logger.info(f"__collect_if_mib_fields: {fields}")
+        return fields
 
     def __enrich_if_mib(self, metric_name):
         result = []
-        if metric_name and metric_name.startswith(InterfaceMib.IF_MIB_METRIC_PREFIX):
-            if (
-                self._mib_static_data_collection
-                and "IF-MIB" in self._mib_static_data_collection
-            ):
-                for dimension in self._mib_static_data_collection["IF-MIB"]:
+        if metric_name and metric_name.startswith(InterfaceMib.IF_MIB_METRIC_SUFFIX):
```

Comment:
I think this is incorrectly merged as we should have PREFIX after resolving conflicts.